### ISSUE TITLE
Android zoom issues

### ISSUE THREAD

**Issue by AlexVincent525** [pichillilorenzo/flutter_inappwebview#270]
## Environment

**Flutter version:** 1.12.13+hotfix.5
**Plugin version:**  3.0.0 9c7ac0d
**Android version:**  Android 8/ 10 
**Device information:**  Emulator, OnePlus 3, OnePlus 6T

## Description

There're **AT LEASE 3 ISSUES** related to zoom control on Android platform. (#232, #258, etc)
As I investigate, both of them can be solved with some imperfect solution. I'll explain them below.
For now, zoom on Android is nearly broken.

### Step to reproduce

1. Set `displayZoomControls` with `true`.
2. Click/Tap on control buttons, nothing happened.

### Root cause

The reason why pinch gesture failed infected by two reasons.

1. `setDesktopMode` re-write user defined options.
https://github.com/pichillilorenzo/flutter_inappwebview/blob/9c7ac0da8f01ceb40430da58397bdb31968ca5b1/android/src/main/java/com/pichillilorenzo/flutter_inappwebview/InAppWebView/InAppWebView.java#L653-L663
Default value of `preferredContentMode` is `RECOMMENDED (0)`, when the code run into this function, it will disabled `loadWithOverviewMode`, `useWideViewPort`, `builtInZoomControls`, `supportZoom` no matter what value you set before.
2. When `displayZoomControls` is `false` which is by default, the pinch gesture is failed but double tap still worked. I didn't found why this will happened.

### Solution

So the solution comes clearly.
1. Lines at `android/src/main/java/com/pichillilorenzo/flutter_inappwebview/InAppWebView/InAppWebView.java#L653-L663` can be changed to:
```
if (options.preferredContentMode == PreferredContentModeOptionType.DESKTOP.toValue()) {
    setDesktopMode(true);
}
```
Then split user agent switcher from `setDesktopMode` for a better UA creation.
2. Figure out why `displayZoomControls` takes effect to zoom gestures.

**Comment by ixen-hj** [comment]
> When displayZoomControls is false which is by default, the pinch gesture is failed but double tap still worked. I didn't found why this will happened.

Seems like the following function is the cause of zoom not working if you set `displayZoomControls` to false

`setOnTouchListener(new View.OnTouchListener() {
      float m_downX;
      float m_downY;

      @Override
      public boolean onTouch(View v, MotionEvent event) {

        if (event.getPointerCount() > 1) {
          //Multi touch detected
          return true;
        }

        if (options.disableHorizontalScroll && options.disableVerticalScroll) {
          return (event.getAction() == MotionEvent.ACTION_MOVE);
        }
        else if (options.disableHorizontalScroll || options.disableVerticalScroll) {
          switch (event.getAction()) {
            case MotionEvent.ACTION_DOWN: {
              // save the x
              m_downX = event.getX();
              // save the y
              m_downY = event.getY();
              break;
            }
            case MotionEvent.ACTION_MOVE:
            case MotionEvent.ACTION_CANCEL:
            case MotionEvent.ACTION_UP: {
              if (options.disableHorizontalScroll) {
                // set x so that it doesn't move
                event.setLocation(m_downX, event.getY());
              } else {
                // set y so that it doesn't move
                event.setLocation(event.getX(), m_downY);
              }
              break;
            }
          }
        }
        return false;
      }
    });`

By commenting out the function. I was able to zoom the page without showing zoom controls.

Tested on a Xiaomi Redmi Note 4 - Android 6
Flutter version: 1.9.1+hotfix5
Plugin version: 2.1.0+1

**Comment by AlexVincent525** [comment]
> Seems like the following function is the cause of zoom not working if you set `displayZoomControls` to false
Huge thanks for pointing it out, I'll try this ASAP.

**Comment by pichillilorenzo** [comment]
Published new version `3.0.0` with the fix! Thanks!
